sqlmesh/core/model/definition.py

```diff
@@ -926,6 +1043,22 @@ def _query_renderer(self) -> QueryRenderer:
             )
         return self.__query_renderer
 
+    @property
+    def _additional_data_for_hash(self) -> t.List[str]:
+        query = self.query if self.hash_raw_query else self.render_query() or self.query
+        data = [query.sql(comments=False), *super()._additional_data_for_hash]
+
+        for macro_name, macro in sorted(self.jinja_macros.root_macros.items(), key=lambda x: x[0]):
```

Comment:
i don't know if you need key here since by default it will sort the tuple by the first  element